packages/node_modules/@ciscospark/widget-recents/src/components/UserProfileAvatar.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import {Avatar, MenuContent, Menu, MenuItem, MenuOverlay} from '@collab-ui/react';
+
+import PresenceAvatar from '@ciscospark/react-container-presence-avatar';
+
+import styles from './UserProfileAvatar.css';
+
+
+const propTypes = {
+  currentUser: PropTypes.shape({
+    id: PropTypes.string,
+    displayName: PropTypes.string,
+    email: PropTypes.string,
+    img: PropTypes.string
+  }).isRequired,
+  enableUserProfile: PropTypes.bool.isRequired,
+  enableUserProfileMenu: PropTypes.bool.isRequired,
+  onSignOutClick: PropTypes.func,
+  onProfileClick: PropTypes.func
+};
+
+const defaultProps = {
+  onSignOutClick: () => {},
+  onProfileClick: () => {}
+};
+
+function UserProfileAvatar(props) {
+  const {
+    currentUser,
+    enableUserProfile,
+    enableUserProfileMenu,
+    onProfileClick,
+    onSignOutClick
+  } = props;
+  const presenceAvatar = (
+    <PresenceAvatar
+      avatarId={currentUser.id}
+      name={currentUser.displayName}
+      onClick={onProfileClick}
+    />);
+  const menu = (
+    <MenuOverlay
+      direction="bottom-left"
+      menuTrigger={presenceAvatar}
+      targetOffset={{vertical: 4}}
+    >
+      <MenuContent>
+        <div className={classNames('ciscospark-recents-header-profile-menu', styles.menu)}>
+          <Avatar
+            src={currentUser.img}
+            title={currentUser.displayName}
+            size={84}
+          />
+          <div className={classNames('ciscospark-recents-header-profile-menu-info', styles.info)}>
+            <h4>{currentUser.displayName} </h4>
+            <h5>{currentUser.email}</h5>
+          </div>
+        </div>
+      </MenuContent>
+      <Menu>
+        <MenuItem
+          className={classNames('ciscospark-recents-header-profile-menu-signout', styles.signout)}
+          onClick={onSignOutClick}
+          label="Sign Out"
+        />
+      </Menu>
+    </MenuOverlay>
+  );
+
+  const profileAvatar = enableUserProfile ? presenceAvatar : null;
```

Comment:
Is this check necessary? We throw an error if `enableUserProfileMenu` is enabled but `enableUserProfile` isn't.

My suggestion would be to leave the if check in `RecentsHeader.js` for the avatar, and only check for the menu prop here.

```js
      {
          enableUserProfile &&
          <UserProfileAvatar
```

Reply:
discussing about it in teams.